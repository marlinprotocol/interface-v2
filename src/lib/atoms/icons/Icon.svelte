<script lang="ts">
	import Icon from 'svelte-awesome/components/Icon.svelte';

	export let data: any;
	export let size = 16;
	export let padding = 1.5;
	export let flip: 'horizontal' | 'vertical' | undefined = undefined;
	export let spin = false;
	export let border = false;
	export let iconColorClass = 'icon-info';

	export let styleClass = '';

	$: baseClass = `width:${size}px; height:${size}px; cursor:pointer;`;
	$: style = `${baseClass} ${
		border ? 'border: 1px solid; border-radius: 20px; padding:' + padding + 'px' : ''
	} ${styleClass}`;
</script>

<Icon {data} {flip} {spin} {style} class={iconColorClass} data-testid="icon" />
