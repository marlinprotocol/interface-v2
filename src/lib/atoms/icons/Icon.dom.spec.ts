import { render } from '@testing-library/svelte';
import Icon from './Icon.svelte';
import { describe, it, afterEach } from 'vitest';
import { cleanup } from '@testing-library/svelte';
import chevronDown from 'svelte-awesome/icons/chevronDown';

describe('IconWrapper', () => {
	afterEach(cleanup);
	const testSize = 20;
	const testPadding = 2;

	it('renders the Icon component', () => {
		const { container } = render(Icon, { data: chevronDown });
		expect(container).toMatchSnapshot();
	});

	it('applies dynamic style based on "size", "padding", and "border" props', () => {
		const { getByTestId } = render(Icon, {
			props: { size: testSize, padding: testPadding, border: true, data: chevronDown }
		});
		const icon = getByTestId('icon');
		const expectedStyle = `width: ${testSize}px; height: ${testSize}px; cursor: pointer; border: 1px solid; border-radius: 20px; padding: ${testPadding}px;`;
		expect(icon.style.cssText).toBe(expectedStyle);
	});

	it('applies dynamic style based on "size", "padding"', () => {
		const { getByTestId } = render(Icon, {
			props: { size: testSize, padding: testPadding, border: false, data: chevronDown }
		});
		const icon = getByTestId('icon');
		// padding gets ignored if border is false.
		const expectedStyle = `width: ${testSize}px; height: ${testSize}px; cursor: pointer;`;
		expect(icon.style.cssText).toBe(expectedStyle);
	});

	it('applies "flip" prop to the Icon', () => {
		const testFlip = 'horizontal';
		const { getByTestId } = render(Icon, { flip: testFlip, data: chevronDown });
		const icon = getByTestId('icon');
		expect(icon.getAttribute('class')).contains('fa-flip-horizontal');
	});

	it('applies "spin" prop to the Icon', () => {
		const { getByTestId } = render(Icon, { spin: true, data: chevronDown });
		const icon = getByTestId('icon');
		expect(icon.getAttribute('class')).contains('fa-spin');
	});

	it('adds "iconColorClass" to the class list of the Icon', () => {
		const testIconColorClass = 'text-blue-500';
		const { getByTestId } = render(Icon, { iconColorClass: testIconColorClass, data: chevronDown });
		const icon = getByTestId('icon');
		expect(icon.getAttribute('class')).toContain(testIconColorClass);
	});

	it('includes "styleClass" in the class list of the Icon', () => {
		const testStyleClass = 'custom-icon-class';
		const { getByTestId } = render(Icon, { styleClass: testStyleClass, data: chevronDown });
		const icon = getByTestId('icon');
		expect(icon.getAttribute('style')).toContain(testStyleClass);
	});
});
