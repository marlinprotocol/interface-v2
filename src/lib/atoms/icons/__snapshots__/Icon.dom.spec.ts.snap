// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconWrapper > renders the Icon component 1`] = `
<div>
  <svg
    aria-label=""
    class="fa-icon icon-info svelte-1mc5hvj"
    data-testid="icon"
    height="16"
    role="presentation"
    style="width:16px; height:16px; cursor:pointer;  "
    version="1.1"
    viewBox="0 0 1792 1792"
    width="16"
  >
    <path
      d="M1683 808l-742 741q-19 19-45 19t-45-19l-742-741q-19-19-19-45.5t19-45.5l166-165q19-19 45-19t45 19l531 531 531-531q19-19 45-19t45 19l166 165q19 19 19 45.5t-19 45.5z"
    />
     
     
    
  </svg>
</div>
`;
